pkg/runtime/grpc/grpc-runtime.go

```diff
@@ -375,6 +375,10 @@ func (r *Runtime) dialContext(dialCtx context.Context, target target, timeout ti
 			gadgetNamespace := r.globalParams.Get(ParamGadgetNamespace).AsString()
 			return NewK8SPortFwdConn(ctx, r.restConfig, gadgetNamespace, target, port, timeout)
 		}))
+	} else {
```

Comment:
Why it doesn't apply to all cases?

Reply:
Good question - I'd have to check again to give a proper answer - I know there were differences in how contexts were handled/used (e.g.: only for connection setup or for the actual outgoing connections), since the grpc connections have many underlying options/paths (like failovers, load balancing and such.)

Reply:
Good thanks. Feel free to merge it if you're happy with it.

Reply:
Yeah, I'll merge for now as this improves the situation for `ig` already.